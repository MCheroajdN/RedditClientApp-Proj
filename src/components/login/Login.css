#login-btn {
	border-radius: 40px;
	padding: 0.3rem 1rem;
	font-size: 15px;
	font-weight: bold;
	margin: 0 0.4rem;
}

#username,
#password {
	font-size: 0.6rem;
	background-color: #d2f4f5;
	opacity: 0.5;
}

#loginFormModal {
	margin-top: 3rem;
}

.modal-backdrop {
	z-index: 4;
}
@media (max-width: 600px) {
	#login-btn {
		display: none;
	}
}
