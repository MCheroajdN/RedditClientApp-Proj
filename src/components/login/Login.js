import React from 'react';
import './Login.css';
import LoginForm from './LoginForm';

const Login = () => {
	return (
		<div>
			<button
				type='button'
				id='login-btn'
				className='btn border-primary bg-light text-primary'
				data-target='#loginFormModal'
				data-toggle='modal'
				aria-label='login button'>
				Login
			</button>

			<LoginForm />
		</div>
	);
};

export default Login;
