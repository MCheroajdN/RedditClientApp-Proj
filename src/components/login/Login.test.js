import { BrowserRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Login from './Login';

describe('Login component', () => {
	it('should have Login', () => {
		const wrapper = shallow(<Login />);
		const login = wrapper.find("button[data-target='#loginFormModal']").text();
		expect(login).toEqual('Login');
	});

	it('matches snapshot', () => {
		const tree = renderer
			.create(
				<Router>
					<Login />
				</Router>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
