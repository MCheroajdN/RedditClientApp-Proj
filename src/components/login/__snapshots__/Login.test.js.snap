// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matches snapshot 1`] = `
<div>
  <button
    aria-label="login button"
    class="btn border-primary bg-light text-primary"
    data-target="#loginFormModal"
    data-toggle="modal"
    role="button"
    type="button"
  >
    Login
  </button>
  <div
    className="modal fade"
    id="loginFormModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8 offset-2"
      >
        <div
          className="modal-dialog"
          role="document"
        >
          <div
            className="modal-content text-left"
          >
            <div
              className="modal-header"
            >
              <div
                className="modal-title  pt-4"
              >
                Login
                <small
                  className="d-block"
                >
                  By continuing, you agree to our
                   
                  <span
                    className="text-primary"
                  >
                    User Agreement 
                  </span>
                   and
                   
                  <span
                    className="text-primary"
                  >
                    Privacy Policy
                  </span>
                  .
                </small>
              </div>
              <button
                aria-label="Close"
                className="close"
                data-dismiss="modal"
                type="button"
              >
                 
                <span
                  arial-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </div>
            <div
              className="modal-body pt-5"
            >
              <div
                className="border border-primary  p-3"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  <i
                    className="fab fa-google"
                  />
                   Continue With Google
                </a>
              </div>
              <div
                className="border border-primary  p-3 mt-2"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  <i
                    className="fab fa-apple"
                  />
                   Continue With Apple
                </a>
              </div>
              <p
                className="text-muted ml-5 pt-3"
              >
                OR 
              </p>
              <div>
                <form
                  className="pt-3 "
                >
                  <input
                    className=" text-muted p-3 form-control"
                    id="username"
                    name="username"
                    placeholder="USERNAME"
                    type="text"
                  />
                  <input
                    className=" text-muted p-3 mt-2 form-control"
                    id="password"
                    name="password"
                    placeholder="PASSWORD"
                    type="password"
                  />
                  <input
                    className="bg-primary mt-2 text-light rounded form-control"
                    type="submit"
                    value="login"
                  />
                </form>
                <small
                  className="mt-3"
                >
                  Forgot your
                   
                  <a
                    className="text-primary"
                    href="/"
                    onClick={[Function]}
                  >
                    username
                  </a>
                   
                  or
                   
                  <a
                    className="text-primary"
                    href="/"
                    onClick={[Function]}
                  >
                    password
                  </a>
                   
                  ?
                </small>
                <small
                  className="mt-3 d-block"
                >
                  New to Reddit?
                  <span
                    className="text-primary"
                    data-dismiss="modal"
                    data-target="#signUpFormModal"
                    data-toggle="modal"
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                     
                    SIGN UP
                  </span>
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
